lib/pdf_fill/processors/base_form_4142_processor.rb

```diff
@@ -0,0 +1,175 @@
+    def generate_stamp_pdf
+      selected_form_class_id = generate_2024_version? ? FORM_CLASS_ID_2024 : LEGACY_FORM_CLASS_ID
+
+      pdf = PdfFill::Filler.fill_ancillary_form(
+        form_data,
+        pdf_identifier,
+        selected_form_class_id
+      )
+
+      # Handle signature stamping for 2024 version
+      if signature.present? && selected_form_class_id == FORM_CLASS_ID_2024
+        pdf = PDFUtilities::DatestampPdf.new(pdf).run(
+          text: signature,
+          x: 50,
+          y: 560,
+          text_only: true,
+          size: 10,
+          page_number: 1,
+          template: pdf,
+          multistamp: true,
+          timestamp_required: false
+        )
+      end
+
+      # Add VA.gov timestamp
+      stamped_path = PDFUtilities::DatestampPdf.new(pdf).run(
+        text: 'VA.gov',
+        x: 5,
+        y: 5,
+        timestamp: submission_date
+      )
+
+      # Add VA.gov Submission text
+      stamped_path = PDFUtilities::DatestampPdf.new(stamped_path).run(
+        text: 'VA.gov Submission',
+        x: 510,
+        y: 775,
+        text_only: true
+      )
+    end
+
+    protected
+
+    # Default: use legacy form
+    def generate_2024_version?
+      false
+    end
+    # Abstract methods to be implemented by subclasses
+    def form_data
+      raise NotImplementedError, 'Subclasses must implement form_data method'
+    end
+
+    def pdf_identifier
+      raise NotImplementedError, 'Subclasses must implement pdf_identifier method'
+    end
+
+    def metadata_uuid
+      raise NotImplementedError, 'Subclasses must implement metadata_uuid method'
+    end
+
+    def submission_date
+      raise NotImplementedError, 'Subclasses must implement submission_date method'
+    end
+
+    def country_code_for_us_validation
+      'US' # Default implementation, can be overridden
+    end
+
+    private
+
+    def uuid
+      @uuid ||= SecureRandom.uuid
+    end
+
+    def to_faraday_upload
+      Faraday::UploadIO.new(
+        @pdf_path,
+        Mime[:pdf].to_s
+      )
+    end
+
+    def generate_metadata
+      address = form_data['veteranAddress']
+      country_is_us = address['country'] == country_code_for_us_validation
+      veteran_full_name = form_data['veteranFullName']
+
+      metadata = {
+        'veteranFirstName' => veteran_full_name['first'],
+        'veteranLastName' => veteran_full_name['last'],
+        'fileNumber' => form_data['vaFileNumber'] || form_data['veteranSocialSecurityNumber'],
+        'receiveDt' => received_date,
+        'uuid' => metadata_uuid,
+        'zipCode' => address['postalCode'],
+        'source' => 'VA Forms Group B',
+        'hashV' => Digest::SHA256.file(@pdf_path).hexdigest,
+        'numberAttachments' => 0,
+        'docType' => FORM_SCHEMA_ID,
+        'numberPages' => PDF::Reader.new(@pdf_path).pages.size
+      }
+
+      SimpleFormsApiSubmission::MetadataValidator.validate(
+        metadata, zip_code_is_us_based: country_is_us
```

Comment:
Then you could just call the helper method here like:
`zip_code_is_us_based: country_code_is_us?(address['country'])`